apps/openmw/mwmechanics/aiwander.cpp

```diff
@@ -303,8 +334,55 @@ namespace MWMechanics
 
             if (storage.mPathFinder.isPathConstructed())
             {
-                storage.mState = Wander_Walking;
+                storage.setState(Wander_Walking);
+            }
+        }
+    }
+
+    /*
+     * Commands actor to walk to a random location near original spawn location.
+     *
+     * Creatures simply wander a certain distance from their starting location, while NPCs wander a scripted
+     * distance (mDistance) from the position where they started the wander package.
+     * http://www.uesp.net/wiki/Tes3Mod:AIWander
+     */
+    void AiWander::wanderNearStart(const MWWorld::Ptr &actor, AiWanderStorage &storage, int wanderDistance, bool isNpc) {
+        const ESM::Pathgrid::Point currentPosition = actor.getRefData().getPosition().pos;
+
+        // Determine a random location within radius of original position
+        const float pi = 3.14159265359f;
+        const float maxWanderDistance = isNpc ? wanderDistance : MINIMUM_WANDER_DISTANCE * 14.0f;
+        const float wanderRadius = Misc::Rng::rollClosedProbability() * maxWanderDistance;
+        const float randomDirection = Misc::Rng::rollClosedProbability() * 2.0f * pi;
+        const float destinationX = mInitialActorPosition.x() + wanderRadius * std::cos(randomDirection);
+        const float destinationY = mInitialActorPosition.y() + wanderRadius * std::sin(randomDirection);
```

Comment:
I don't know how sophisticated the original engine's code is in this respect, but I don't think the above code gives a uniform distribution on a circle as intended. A higher radius ought to be more likely than a small radius, since it covers more area. As usual, SO has the correct answer: http://stackoverflow.com/questions/5837572/generate-a-random-point-within-a-circle-uniformly